src/openrct2/ride/RideData.cpp

```diff
@@ -320,18 +320,16 @@ bool RideTypeDescriptor::HasFlag(uint64_t flag) const
     return Flags & flag;
 }
 
-uint64_t RideTypeDescriptor::GetAvailableTrackPieces() const
+void RideTypeDescriptor::GetAvailableTrackPieces(RideTrackGroup& res) const
 {
+    res |= EnabledTrackPieces;
     if (gCheatsEnableAllDrawableTrackPieces)
-    {
-        return EnabledTrackPieces | ExtraTrackPieces;
-    }
-    return EnabledTrackPieces;
+        res |= ExtraTrackPieces;
 }
 
 bool RideTypeDescriptor::SupportsTrackPiece(const uint64_t trackPiece) const
 {
-    return GetAvailableTrackPieces() & (1ULL << trackPiece);
+    return EnabledTrackPieces.get(trackPiece) || (gCheatsEnableAllDrawableTrackPieces && ExtraTrackPieces.get(trackPiece));
```

Comment:
What is the point of this, considering `GetAvailableTrackPieces()` already takes care of checking `gCheatsEnableAllDrawableTrackPieces`?